### PR TITLE
added no_proxy + use-authentication prop

### PR DESCRIPTION
**Pull request by tuxfy**

### CODE REVIEW

File: gsettings.sh [hunk 1]
```diff
@@ -38,12 +44,31 @@ set_proxy() {
...
+    if [[ $no_proxy =~ .*,.* ]]; then
+        IFS=',' read -r -a array <<< "$no_proxy"
+        _no_proxy=$(printf ",'%s'" "${array[@]}")
+        _no_proxy="[${_no_proxy:1}]"
+        gsettings set org.gnome.system.proxy ignore-hosts "${_no_proxy}"
+    elif [[ $no_proxy != "" ]]; then
+            gsettings set org.gnome.system.proxy ignore-hosts "['${no_proxy}']"
+    else
+        gsettings set org.gnome.system.proxy ignore-hosts ""
+    fi
```
Comment (himanshub16) [inline gsettings.sh]: This one seems helpful!

File: gsettings.sh [hunk 2]
```diff
@@ -27,6 +27,12 @@ list_proxy() {
     echo "${bold} socks ${normal} "\
          "$(gsettings get org.gnome.system.proxy.socks host) "\
          "$(gsettings get org.gnome.system.proxy.socks port)"
+    echo "${bold} auth ${normal} "\
+         "$(gsettings get org.gnome.system.proxy.http use-authentication) "\
+         "$(gsettings get org.gnome.system.proxy.http authentication-user) "\
```
[the remark this answers is not in the document]
Comment (tuxfy) [reply gsettings.sh]: Oh, sorry :)

File: gsettings.sh [hunk 3]
```diff
@@ -38,12 +44,31 @@ set_proxy() {
...
-    gsettings set org.gnome.system.proxy.http authentication-user "$username"
     gsettings set org.gnome.system.proxy.https host "$https_host"
     gsettings set org.gnome.system.proxy.https port "$https_port"
     gsettings set org.gnome.system.proxy.ftp host "$ftp_host"
     gsettings set org.gnome.system.proxy.ftp port "$ftp_port"
+
+    if [ "$use_auth" = "y" ]; then
+        gsettings set org.gnome.system.proxy.http use-authentication true
+        gsettings set org.gnome.system.proxy.http authentication-password "$password"
+        gsettings set org.gnome.system.proxy.http authentication-user "$username"
```
[the remark this answers is not in the document]
Comment (tuxfy) [reply gsettings.sh]: cleaned

File: gsettings.sh [hunk 4]
```diff
@@ -26,7 +27,9 @@ list_proxy() {
          "$(gsettings get org.gnome.system.proxy.ftp port)"
     echo "${bold} socks ${normal} "\
          "$(gsettings get org.gnome.system.proxy.socks host) "\
-         "$(gsettings get org.gnome.system.proxy.socks port)"
+         "$(gsettings get org.gnome.system.proxy.socks port)"
```
Comment (himanshub16) [inline gsettings.sh]: Why has this line changed? 
Perhaps due to `tab` and `space`.

Comment (tuxfy) [reply gsettings.sh]: I don't know.  Didn't change anything here. Maybe some intendation stuff from IDE?

**Pull request merged**